addons/web/static/src/views/graph/graph_renderer.js

```diff
@@ -351,6 +352,11 @@ export class GraphRenderer extends Component {
             dataset.fill = "origin";
             dataset.backgroundColor = hexToRGBA(getColor(0), 0.4);
         }
+        if (stacked) {
+            for (const dataset of data.datasets) {
+                dataset.backgroundColor = hexToRGBA(dataset.borderColor, 0.4);
+            }
+        }
```

Comment:
While this is probably easier to read, it seems that you just loop over dataset a second time for "nothing".
Maybe check if you can put this in the for loop above, something like:
```js
if (stacked) {
    dataset.backgroundColor = hexToRGBA(dataset.borderColor, 0.4);
}
```
I ultimately agree that this code is somewhat gross.